page-optimize.php

```diff
@@ -54,9 +64,14 @@ function page_optimize_cache_cleanup( $file_age = DAY_IN_SECONDS ) {
 
 // Unschedule cache cleanup, and purge cache directory
 function page_optimize_deactivate() {
-	page_optimize_cache_cleanup( 0 /* max file age */ );
+	$cache_folder = false;
+	if ( defined( 'PAGE_OPTIMIZE_CACHE_DIR' ) && ! empty( PAGE_OPTIMIZE_CACHE_DIR ) ) {
+		$cache_folder = PAGE_OPTIMIZE_CACHE_DIR;
+	}
+
+	page_optimize_cache_cleanup( $cache_folder, 0 /* max file age in seconds */ );
 
-	wp_clear_scheduled_hook( PAGE_OPTIMIZE_CRON_CACHE_CLEANUP_JOB );
+	wp_clear_scheduled_hook( PAGE_OPTIMIZE_CRON_CACHE_CLEANUP_JOB, [ $cache_folder ] );
```

Comment:
Don't think this will work. Seems that it uses the `$args` to uniquely define the jobs: https://developer.wordpress.org/reference/functions/wp_unschedule_event/

Tho it's not a big problem, when the even runs, there will be no actions attached to it, so kinda fine.

Reply:
Actually, I'm wrong. Clearing the cleanup job won't work without matching args.

I'm not sure if there is a good way to iterate through scheduled cleanup events and run them. WP-CLI can list events with their args, but I haven't yet found a WP function to do the same.

Reply:
Yeah, don't really think it matters. It runs daily, when it runs there will be no actions, so no damage done.

Reply:
And it's kinda an edge case.

Reply:
Oops, I didn't notice your "Don't think this will work" before replying with "Actually, I'm wrong". Either way, we're agreed.